IdentityCore/src/util/ios/MSIDAppExtensionUtil.m

```diff
@@ -77,7 +77,11 @@ + (void)sharedApplicationOpenURL:(NSURL*)url
 #pragma clang diagnostic ignored "-Warc-performSelector-leaks"
     
     [MSIDMainThreadUtil executeOnMainThreadIfNeeded:^{
+#if defined TARGET_OS_VISION && TARGET_OS_VISION
```

Comment:
nit: this check could be defined as a new macro something like MSID_TARGET_OS_VISION, so you could avoid doing this && operation everywhere

Reply:
for my own learning, why do we need `#if defined TARGET_OS_VISION && TARGET_OS_VISION` but not `#if TARGET_OS_VISION`? :)

Reply:
I previously tested these changes on an older version of XCode before the visionOS simulator was introduced, and I saw build errors due to using an undefined macro. The "defined" check ensures that we can still build on XCode <15.2